### PR TITLE
Formatting in tensor mechanic docs updates

### PR DESCRIPTION
**Pull request by sapitts**
### Documentation Update

This merge request makes minor changes to the formatting of documentation in tensor mechanics module for equations, headings, and includes. Additional warning documentation to explain why symmetry must be maintained about the Y/Z (not X) axis in the tensor mechanics axisymmetric codes is added.

### Design Changes
The last commit in this pull request moves the doxygen like documentation for the RankTwoTensor and RankFourTensor classes from tensor mechanics to framework_development/moose_utils. It's easy to drop this commit if this documentation move doesn't best serve our users.

@aeslaughter or @bwspenc please review

Refs #10516

### CODE REVIEW

File: framework/doc/content/documentation/framework_development/moose_utils.md [hunk 1]
```diff
@@ -75,3 +75,124 @@ After the data is read using the "read" method, there are two methods used for a
...
+
+- `zero()`
+- `print()`
+- `L2norm()`
+- `invSymm()` returns A$_{ijkl}$ from the input C$_{ijkl}$ such that $C_{ijkl} \cdot A_{ijkl} = 0.5 \left( \delta_{im} \delta_{jn} + \delta_{in} \delta_{jm} \right)$
+- `rotate(RealTensorValue)`
+- `rotate(RankTwoTensor)`
+- `transposeMajor()` witch first pair and second pair of indices
+- `surfaceFillFromInputVector(TypeVector<Real>)` Fills tensor entries by ignoring the last dimension
+- `fillFromInputVector(TypeVector<Real>, FillMethod)` used to build elasticity tensors, see [ComputeElasticityTensor](/Materials/ComputeElasticityTensor.md)
```
Comment (sapitts) [inline framework/doc/content/documentation/framework_development/moose_utils.md]: @aeslaughter I'm at a loss for how to properly link this file. Would you tell me the magic combination for linking across different modules and/or the framework and a module please?

File: modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/StressDivergence.md [hunk 2]
```diff
@@ -1,6 +1,43 @@
...
+\mathbf{\epsilon}_0)$, where $\mathbf{\epsilon}$ is the strain and $\mathbf{\epsilon}_0$ is a stress
+free strain. For example, in linear elasticity (only valid for small strains), the material response
+is linear, i.e.  $\mathbf{\sigma} = \mathbf{\mathcal{C}}(\mathbf{\epsilon} - \mathbf{\epsilon}_0)$.
+
+## Consistency Between Stress and Strain
+
+Within the tensor mechanics module, we have three separate ways to consistently
+calculate the strain and stress as shown in [strain_formulations]. Attention to
+the correspondence among the stress and strain formulations is necessary to ensure
+the stress and strain calculations are performed on the same material configuration.
```
Comment (bwspenc) [inline modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/StressDivergence.md]: on the same material configuration -> on the correct material configurations

(because they aren't necessarily the same)

File: modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/Stresses.md [hunk 3]
```diff
@@ -135,12 +116,13 @@ classes of UserObjects are as follows:
 - `HEVPStrengthUOBase`
 
 
-## Creep and Plasticity with StressUpdate Materials
+## StressUpdate Materials
+
+A set of plasticity and creep material models have been developed with the `StressUpdateBase` class,
+which allows for iterations within the material itself.These iterative
```
Comment (bwspenc) [inline modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/Stresses.md]: itself.These -> itself. These

File: modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/Stresses.md [hunk 4]
```diff
@@ -135,12 +116,13 @@ classes of UserObjects are as follows:
...
+materials are designated by `StressUpdate` at the end of the class name.
 
-In addition to the User Object based plasticity models, another set of plasticity and creep material
-models have been developed with the `StressUpdateBase` class, which allows for iterations within the
-material itself.  These iterative materials are designated by `StressUpdate` at the end of the class
-name. These classes use a Radial Return von Mises, or J2, approach to determine the inelastic, creep,
+These classes use a Radial Return von Mises, or J2, approach to determine the inelastic, creep,
```
Comment (bwspenc) [inline modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/Stresses.md]: These classes can use a variety of approaches to determine the inelastic strain increment at each step. 

(they don't just have to be J2)

File: modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/index.md [hunk 5]
```diff
@@ -1,54 +1,29 @@
...
-assuming $\mathbf{\sigma}_0$ is independent of the strain.
+- [Visualizing Tensors](tensor_mechanics/VisualizingTensors.md)
+- [Tensor Classes](tensor_mechanics/TensorClasses.md)
 
-The material stress response is described by the constitutive model, where the stress is determined
-as a function of the strain, i.e. $\tilde{\mathbf{\sigma}}( \mathbf{\epsilon} -
-\mathbf{\epsilon}_0)$, where $\mathbf{\epsilon}$ is the strain and $\mathbf{\epsilon}_0$ is a stress
-free strain. For example, in linear elasticity (only valid for small strains), the material response
-is linear, i.e.  $\mathbf{\sigma} = \mathbf{\mathcal{C}}(\mathbf{\epsilon} - \mathbf{\epsilon}_0)$.
+The MOOSE +Tensor Mechanics+ module is a library for simulation tools that solve
```
Comment (bwspenc) [inline modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/index.md]: This material seems like it should go at the top of this page.

File: modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/common/supplementalAxisymmetricRZStrain.md [hunk 6]
```diff
@@ -1,14 +1,22 @@
 Axisymmetric (cylindrical) materials are included in Tensor Mechanics for revolved geometries and
 assume symmetrical loading. These 'strain calculator' materials compute the strain within the
 appropriate coordinate system and rely on specialized AxisymmetricRZ kernels to handle the stress
-divergence. The ```AxisymmetricRZ``` material is appropriate for a 2D simulation and assumes symmetry
-revolved about the z-axis in the polar angle. A 2D formulation of an appropriate simulation problem
+divergence.
+
+!alert warning title=Symmetry Assumed About the $z$-axis
+The calculation of the residual and of the jacobian require the axis of symmetry lies along
```
Comment (bwspenc) [inline modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/common/supplementalAxisymmetricRZStrain.md]: I'd reword this as:
The axis of symmetry must lie along the $z$-axis...

**Review by bwspenc — CHANGES_REQUESTED** [review]

File: modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/StressDivergence.md (hunk 2 above)
Comment (sapitts) [reply modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/StressDivergence.md]: Thank you for catching this--I could really confused some users

**Comment by sapitts** [comment]
Thanks for the review @bwspenc; I think I have all of your comments addressed.

**Review by bwspenc — APPROVED** [review]
